pkg/backend/apply.go

```diff
@@ -125,6 +125,12 @@ func PreviewThenPrompt(ctx context.Context, kind apitype.UpdateKind, stack Stack
 		return changes, nil
 	}
 
+	if !changes.HasChanges() {
+		close(eventsChannel)
+
+		os.Exit(0)
+	}
```

Comment:
i'm not sure i like the idea of our code just calling Exit as it's going along.  Can we not just return gracefully and have hte app exit naturally?

Reply:
also, see comment below.  This actually isn't ok given our preview/update model.  We *do* have to let the user update if they want, even if our preview was unable to detect any changes.